Documentation/manual/kcc-design/customization-user-story.md

```diff
@@ -0,0 +1,138 @@
+**Expanding on ‘Customization User’ Scenarios**
+
+**Example #1: Planar 3D Platformer Character Controller**
+
+————————————————————————————————————————————————————————————————————————
+
+User Story:
+
+As a Customization User with an existing project, I’m interested in having my development studio switch to OpenKCC from a paid asset for the following reasons
+- To take advantage of arbitrary colliders
+- To benefit from wide test coverage and reliable low-level code
+- To contribute to, learn from, and benefit from an Open Source code base
+
+While we are interested in the the higher level features of OpenKCC, as a pre-existing studio we already have our own high level code that enables us to reach a high velocity of output with our designers who are already accustomed to our tools
+
+Ideally, our game architecture programmer will be able to use the API of the KCCMovementEngine to make it compatible with our high level code which already handles things such as state switching, etc.
+
+We could probably be convinced to let KCC handle things like gravity for us if the API were well defined and the customization was robust enough, but for the foreseeable future it’s best to think that we are glued to our custom state solution.
+
+**Use Case Level #1: A simple yet robust character that is driven by custom client-side code making use of the Customization Level API**
+
+————————————————————————————————————————————————————————————————————————
+
+**Example Use Case:** For example, a character controller like this is something we would hope to be able to achieve while handling states, input, and events ourselves:
+https://www.youtube.com/watch?v=ylG72jvo_2s&ab_channel=CaptainRipley
+- In fact I intend to start implementing a character controller with equivalent features, that way we could focus on porting it to Open KCC and developing the KCCMovementEngine API alongside it
```

Comment:
That would be awesome, could we add it as a Sample for the OpenKCC repo or would it be using your custom authoring components you're writing on top of it?